accumulator_updater/programs/accumulator_updater/src/state/accumulator_input.rs

```diff
@@ -0,0 +1,84 @@
+use {
+    crate::{
+        accumulator_acc_seeds_with_bump,
+        AccumulatorUpdaterError,
+    },
+    anchor_lang::prelude::*,
+};
+
+//TODO: implement custom serialization & set alignment
+#[account]
+pub struct AccumulatorInput {
+    pub header: AccumulatorHeader,
+    pub data:   Vec<u8>,
+}
+
+impl AccumulatorInput {
+    pub fn get_initial_size(data: &Vec<u8>) -> usize {
+        AccumulatorHeader::SIZE + 4 + data.len()
+    }
+
+    pub fn new(header: AccumulatorHeader, data: Vec<u8>) -> Self {
+        Self { header, data }
+    }
+
+    pub fn validate_account_info(
```

Comment:
this doesn't seem like the right encapsulation of functionality -- it's a very specific function that can only be used in one place. I think a better function would be something like `derive_pda_for(cpi_caller, base_account, account_schema)` ?